faunadb-java-dsl/src/main/java/com/faunadb/client/query/Fn.java

```diff
@@ -25,13 +25,13 @@ protected T toJson() {
 
   private static final class UnescapedObject extends Unescaped<Map<String, Expr>> {
     private UnescapedObject(Map<String, ? extends Expr> body) {
-      super(ImmutableMap.copyOf(body));
+      super(Collections.unmodifiableMap(body));
```

Comment:
An `unmodifiableMap` prevents us from changing an externally created map, but it doesn't prevent external code from changing the map used to create the `UnescapedObject`. I wonder if we should force a copy in cases like this.

Reply:
Yes, that may be a good idea.

Reply:
Is it really necessary? Because that object is only used internally via `Fn.apply()`

Reply:
Maybe not necessarily here but, certainly at the public interface. With guava we gained immutability safety everywhere. Now we need to find the right place for it.